cwms-data-api/src/main/java/cwms/cda/data/dao/timeseriesprofile/TimeSeriesProfileInstanceDao.java

```diff
@@ -0,0 +1,660 @@
+        Field<Timestamp> startTimeCol = VIEW_TSV2.FIRST_DATE_TIME;
+        Field<Timestamp> dateTimeCol = VIEW_TSV2.DATE_TIME;
+
+        // handle previous flag
+        if (previous) {
+            Timestamp previousDateTime = null;
+            SelectConditionStep<Record1<Timestamp>> prev = dsl.select(max(VIEW_TSV2.DATE_TIME))
+                    .from(VIEW_TSV2)
+                    .where(whereCondition.and(dateTimeCol.lessThan(Timestamp.from(startTime)))
+                            .and(endTimeCol.greaterThan(Timestamp.from(startTime))));
+            previousDateTime = prev.fetchOne().value1();
+            if (previousDateTime != null) {
+                startTime = previousDateTime.toInstant();
+                startInclusive = true;
+            }
+        }
+
+        // handle next flag
+        if (next) {
+            Timestamp nextDateTime = null;
+            SelectConditionStep<Record1<Timestamp>> nex = dsl.select(min(VIEW_TSV2.DATE_TIME))
+                    .from(VIEW_TSV2)
+                    .where(whereCondition.and(dateTimeCol.greaterThan(Timestamp.from(endTime)))
+                            .and(startTimeCol.le(Timestamp.from(endTime))));
+            nextDateTime = nex.fetchOne().value1();
+            if (nextDateTime != null) {
+                endTime = nextDateTime.toInstant();
+                endInclusive = true;
+            }
+        }
+
+        // Add the time windows conditions depending on the inclusive flags
+        if (startInclusive && endInclusive) {
+            whereCondition = whereCondition
+                    .and(VIEW_TSV2.FIRST_DATE_TIME.ge(Timestamp.from(startTime)))
+                    .and(VIEW_TSV2.LAST_DATE_TIME.le(Timestamp.from(endTime)));
+        } else if (!startInclusive && endInclusive) {
+            whereCondition = whereCondition
+                    .and(VIEW_TSV2.FIRST_DATE_TIME.greaterThan(Timestamp.from(startTime)))
+                    .and(VIEW_TSV2.LAST_DATE_TIME.le(Timestamp.from(endTime)));
+        } else if (startInclusive) {
+            whereCondition = whereCondition
+                    .and(VIEW_TSV2.FIRST_DATE_TIME.ge(Timestamp.from(startTime)))
+                    .and(VIEW_TSV2.LAST_DATE_TIME.lessThan(Timestamp.from(endTime)));
+        } else {
+            whereCondition = whereCondition
+                    .and(VIEW_TSV2.FIRST_DATE_TIME.greaterThan(Timestamp.from(startTime)))
+                    .and(VIEW_TSV2.LAST_DATE_TIME.lessThan(Timestamp.from(endTime)));
+        }
+        Condition finalWhereCondition = whereCondition;
+
+        // set semi-final variables for lambda
+        final int recordPageSize = pageSize;
+
+        // Get the total number of records if not already set
+        if (total == null) {
+            SelectConditionStep<Record1<Integer>> count = dsl.select(count(VIEW_TSV2.LOCATION_CODE))
+                    .from(VIEW_TSV2)
+                    .where(finalWhereCondition);
```

Comment:
Count is based on times, not on values

Reply:
Updated to count distinct times, now handles paging as expected